opa/rego/rules/provider_known_vulnerability.rego

```diff
@@ -0,0 +1,32 @@
+# METADATA
+# title: Provider with a Known Vulnerability used
+# description: |-
+#   The build or SCM provider used has a known vulnerability.
+# related_resources:
+# - ref: https://osv.dev/
+#   description: Source Advisory Database
+# custom:
+#   level: warning
+package rules.provider_known_vulnerability
+
+import data.external.provider.advisories
+import data.poutine
+import rego.v1
+
+rule := poutine.rule(rego.metadata.chain())
+
+provider_advisory(provider, provider_version) = advisory if {
+	version := provider_version
+	advisory := advisories[provider][osv_id]
```

Comment:
If I understood correct `provider` will have the value `github` or `gitlab` by the time it gets here.

So `github.com` (SAAS) and `gitlab.com` (SAAS) will both get that too, shouldn't those be fast path ignored completely...

And then for the `details` we'll show `Provider: github` ? that's not super useful if the domain is like `github-on-prem.corp.com`